### ISSUE TITLE
Nested dependencies do not get provided ExecutionContext

### ISSUE THREAD

**Issue by coln** [Urigo/graphql-modules#1448]
I am trying to retrieve an Operation-time context within a nested dependency, however the decorated variable "context" is undefined.

I have a nested dependency tree like so:
application -> module -> class B -> class A -> getName() -> context.injector.get(NameKey)

Inside of singleton class A is the @ExecutionContext() decorator.
Inside of singleton class B is a reference in the constructor to class A which calls function getName()

Here is a codesandbox that demonstrates the issue:
https://codesandbox.io/s/typescript-node-forked-3ueo3?file=/src/index.ts

If you query: `query {
   getName
}`
You should see "Hello, My Name!" which means the context is being injected properly.

If you query: `query {
   getDependencyName
}`
You will see an error, because class B is calling a function from class A which relies on the context which is undefined now.

I am using:
graphql-modules v1.0.0-alpha.2
apollo-server v2.19.0

Any help is appreciated!

**Comment by kamilkisiela** [comment]
Reproduction #1449

**Comment by coln** [comment]
I believe this line is causing the issue, because the top-level dependency is being called by the proxy injector, while the nested dependency is being called by the actual injector.
https://github.com/Urigo/graphql-modules/blob/ba609b9f53eccb88cff89a7e8e5dc6c5c3d872c3/packages/graphql-modules/src/di/injector.ts#L264-L266


Which is called by the proxyFactory in this line:
https://github.com/Urigo/graphql-modules/blob/ba609b9f53eccb88cff89a7e8e5dc6c5c3d872c3/packages/graphql-modules/src/di/injector.ts#L111-L114

I believe passing the newInjector down so the dependency resolver solves the issue, however I'm not sure what ramifications this may have:
```typescript
let newInjector: ReflectiveInjector;
function proxyFactory() {
      return new Proxy(injector.get(provider.key.token, _THROW_IF_NOT_FOUND, newInjector), {
        get(target, propertyKey) {
          if (
            typeof propertyKey !== 'number' &&
            provider.factory.executionContextIn.includes(propertyKey)
          ) {
...
...
private _instantiateProvider(provider: ResolvedProvider, proxy?: ReflectiveInjector): any {
    const factory = provider.factory.factory;

    let deps: any[];
    try {
      deps = provider.factory.dependencies.map((dep) =>
        proxy ? proxy._getByDependency(dep) :
          this._getByDependency(dep)
      );
    } catch (e) {
      if (e.addKey) {
        e.addKey(provider.key);
      }
      throw e;
    }
....
....
```

**Comment by kamilkisiela** [comment]
#1449

**Comment by kamilkisiela** [comment]
We released a new major version (v1.0) of GraphQL Modules.

The new version solves the issue, so I'm closing the thread.
